### DISCUSSION TITLE
Is there any way to clear or force update the secret cache?

### DISCUSSION THREAD

**Discussion by feng0905** [concourse/concourse#8754] [Help & Support]
We are planning to use parameter store as the credential manager since we update those secrets infrequently. 
Therefore, in order to limit the api calls to parameter store, we plan to use a fairly long duration to cache the secrets. 

I am wondering if there is a way we can force update the cache if we have updated secrets in parameter store. 

I couldn't find any docs about it. 
Please suggest if there is any way of clear the caches for secrets.

Thanks.

**Comment by taylorsilva** [comment]
AFAIK there isn't a way to force the refresh. One of the hard problems with caching; knowing when to refresh the cache 😞

**Reply by feng0905** [reply]
thanks for your answer. I was looking at the source code and would like to try adding a fly command to refresh the secret cache. 
Do you think it would be appreciated by others? or is this just us that needs this new feature.

**Reply by taylorsilva** [reply]
I think others would find it useful. I've definitely found myself stuck with a bad secret cache and wishing I could refresh it!
